app/helpers/odk_helper.rb

```diff
@@ -28,6 +29,18 @@ def required_value(form)
     form.allow_incomplete? ? "selected(/data/#{IR_QUESTION}, 'no')" : "true()"
   end
 
+  def appearance(grid)
+    (grid) ? "list-nolabel" : nil
+  end
+
+  # tests if all items in the group are part of the grid
+  def grid_mode?(qings)
+    qings.inject(true) do |memo, q|
+      memo &= q.qtype.name == "select_one"
+      memo
+    end
+  end
```

Comment:
* I think this belongs in the QingGroup model.
* Also, you should be checking both that it's select_one and that the option set is the same.
* `inject` is nice, but here you could just do `qings.all?{ |q| q.qtype.name == 'select_one' && q.option_set == qings[0].option_set` }